src/extension.js

```diff
@@ -4,11 +4,35 @@ console.log("Scroll Percentage in Tab Title Extension loaded 👍");
 
 const originalTitle = document.title;
 
-window.addEventListener("scroll", () => {
-  let scrollTop = window.scrollY;
-  let docHeight = document.body.offsetHeight;
-  let winHeight = window.innerHeight;
-  let scrollPercent = scrollTop / (docHeight - winHeight);
-  let scrollPercentRounded = Math.round(scrollPercent * 100);
-  document.title = `(${scrollPercentRounded}%) ${originalTitle}`;
-}, { passive: true });
+window.addEventListener(
+  "scroll",
+  () => {
+    const windowHeight = document.body.offsetHeight;
+    const scrollTop = window.scrollY;
+    const containers = document.querySelectorAll(
+      "body,main,*[class~='content'],*[class*='Content'],*[class~='article']:first-child,article:first-child"
+    );
+
+    let mainContainer = null,
+      mainContainerSizes = null;
+
+    for (let x = containers.length - 1; x >= 0; x--) {
```

Comment:
I suggest extracting this block to a helper function.
something like `getMainContianer`.

Reply:
I have more feature/refactor branch (here)[https://github.com/dimshik100/Scroll-Percentage-in-Tab-Title-Chrome-Extension/compare/master...Mosharush:feature/use-intersection-observer-instead-scroll-event].
I do something similar to your opinion there, but I checkout from this branch.

Reply:
Rebased to here